src/NServiceBus.AmazonSQS/PreparedMessage.cs

```diff
@@ -1,18 +1,64 @@
 namespace NServiceBus.Transports.SQS
 {
+    using System;
     using System.Collections.Generic;
     using Amazon.SQS.Model;
 
     class PreparedMessage
     {
         public string MessageId { get; set; }
-        public string Body{ get; set; }
+        public string Body { get; set; }
         public string Destination { get; set; }
         public string OriginalDestination { get; set; }
         public string QueueUrl { get; set; }
         public int DelaySeconds { get; set; }
         public Dictionary<string, MessageAttributeValue> MessageAttributes { get; } = new Dictionary<string, MessageAttributeValue>();
         public string MessageGroupId { get; set; }
         public string MessageDeduplicationId { get; set; }
+
+        public long CalculateSize()
+        {
+            long size = Body.Length;
+            foreach (var messageAttributeValue in MessageAttributes)
+            {
+                size += messageAttributeValue.Key.Length;
```

Comment:
From my experiments it looks that _DataType_ counts as well, sending this: 

`	var sendMessageBatchRequest = new SendMessageBatchRequest();

	sendMessageBatchRequest.Entries.Add(new SendMessageBatchRequestEntry(Guid.NewGuid().ToString(), new string('x', 1024 * 256))
	{
		MessageAttributes = new Dictionary<string, MessageAttributeValue>
		{
			["foo"] = new MessageAttributeValue { DataType = "String", StringValue = "abcdef" },
			["bar"] = new MessageAttributeValue { DataType = "String", StringValue = "xyz" }
		},
	});`

Results in _One or more errors occurred. (Batch requests cannot be longer than 262144 bytes. You have sent 262171 bytes.)_, which is 27 bytes above the limit.

Reply:
Interesting. I reread the spec again and saw now

> The data type has the same restrictions as the message body (for more information)

It seems in my first read-through I thought it only applies for custom data types. Will fix